templates/inspec/singular_resource.erb

```diff
@@ -64,7 +64,11 @@ class <%= object.name -%> < GcpResourceBase
     if time?(prop)
       init = "parse_time_string(@fetched['#{prop.api_name}'])"
     elsif primitive?(prop)
-      init = "@fetched['#{prop.api_name}']"
+      if prop.name_from_self_link
+        init = "name_from_self_link(@fetched['#{prop.api_name}'])"
```

Comment:
As you noted in the other PR we will need to wait to merge this until the `name_from_self_link` method is available in gcp_backend.

Alternatively we could put the method definition in the the `singular_resource.erb` template and only add it if a property has the `name_from_self_link` attribute.

Reply:
Sure ! This method will be used on a lot of resources such as `google_compute_zone`, `google_compute_forwarding_rules`, `disks`, ... That why I think interesting to put it as global method.

Reply:
Generally I don't find a lot of value in writing tests against the `name` field on these resources because the name is required to retrieve it. It seems a little redundant to retrieve a resource by name and then assert that the name of the retrieved resource equals that name

But if it is useful I'm happy to add it as a global method